source/index.ts

```diff
@@ -228,15 +231,25 @@ export default class PQueue<QueueType extends Queue<RunFunction, EnqueueOptionsT
 		});
 	}
 
+	async setPriority(uid: string, priority: number) {
+		this.#queue.prioritize(uid, priority);
+	}
+
 	/**
 	Adds a sync or async task to the queue. Always returns a promise.
 	*/
-	async add<TaskResultType>(function_: Task<TaskResultType>, options: {throwOnTimeout: true} & Exclude<EnqueueOptionsType, 'throwOnTimeout'>): Promise<TaskResultType>;
-	async add<TaskResultType>(function_: Task<TaskResultType>, options?: Partial<EnqueueOptionsType>): Promise<TaskResultType | void>;
-	async add<TaskResultType>(function_: Task<TaskResultType>, options: Partial<EnqueueOptionsType> = {}): Promise<TaskResultType | void> {
+	async add<TaskResultType>(function_: Task<TaskResultType>, options: {throwOnTimeout: true} & Exclude<EnqueueOptionsType, 'throwOnTimeout'>, uid?: string): Promise<TaskResultType>;
+	async add<TaskResultType>(function_: Task<TaskResultType>, options?: Partial<EnqueueOptionsType>, uid?: string): Promise<TaskResultType | void>;
+	async add<TaskResultType>(function_: Task<TaskResultType>, options: Partial<EnqueueOptionsType> = {}, uid?: string): Promise<TaskResultType | void> {
```

Comment:
What's the use-case for a custom ID? I would prefer not to expose this unless there are real-world use-cases.

Reply:
shifted the id in `EnqueueOptionsType`. use case is simple a unique identifier that can be assigned to query in the queue.

Reply:
The problem with both accepting an ID and generating is that the user could potentially pass in an ID that equals one of the existing IDs generated internally.

Reply:
we could prevent user to change/assign the id and completely rely on the auto generated id - by idAssigner, but we would need to devise a way to emit this info as soon as the id is generated? could you please suggest on this?